src/Rules/Validation/ScopeValidationMethods.php

```diff
@@ -0,0 +1,156 @@
+<?php declare(strict_types=1);
+
+namespace Hihaho\PhpstanRules\Rules\Validation;
+
+use Illuminate\Support\Collection;
+use Illuminate\Support\Stringable;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\NodeAbstract;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\AnonymousClassNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Type\ObjectType;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionIntersectionType;
+use ReflectionMethod;
+use ReflectionNamedType;
+use ReflectionParameter;
+use ReflectionUnionType;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\MethodCall>
+ */
+abstract class ScopeValidationMethods implements Rule
+{
+    protected array $allowedRequestMethods = [];
+
+    abstract public function getNodeType(): string;
+
+    abstract public function processNode(Node $node, Scope $scope): array;
+
+    /**
+     * @phpstan-return ReflectionMethod[]
+     */
+    protected function getClassMethods(Scope $scope): array
+    {
+        if (! $className = $scope->getClassReflection()?->getName()) {
+            return [];
+        }
+
+        $type = new ObjectType(className: $className, classReflection: $scope->getClassReflection());
+        /** @var ReflectionMethod[] $methods */
+        $methods = array_map(static function (string $className): array {
+            try {
+                return (new ReflectionClass($className))->getMethods();
+            } catch (ReflectionException) {
+                // @phpstan-ignore phpstanApi.constructor
```

Comment:
ignoring because PHPStan complains that it is not backwards compatible, but otherwise we'd get a "not found" for anonymous classes.